cfme/intelligence/reports/widgets/chart_widgets.py

```diff
@@ -34,20 +35,14 @@ class EditChartWidgetView(BaseEditDashboardWidgetView, ChartWidgetFormCommon):
     pass
 
 
+@attr.s
 class ChartWidget(BaseDashboardReportWidget):
 
     TYPE = "Charts"
     TITLE = "Chart"
     pretty_attrs = ["title", "description", "filter", "visibility"]
-
-    def __init__(self,
-            title, description=None, active=None, filter=None, timer=None, visibility=None):
-        self.title = title
-        self.description = description
-        self.active = active
-        self.filter = filter
-        self.timer = timer
-        self.visibility = visibility
+    filter = attr.ib(default=None)
+    timer = attr.ib(default=attr.Factory(dict))
```

Comment:
**Question**

Why change this default to a dict?

Reply:
Because below I fill a dict and I need to get values from `timer` attribute which is suspposed to be a dict:
```python
    @property
    def fill_dict(self):
        return {
            "widget_title": self.title,
            "description": self.description,
            "active": self.active,
            "filter": self.filter,
            "run": self.timer.get("run"),
            "every": self.timer.get("hours"),
            "time_zone": self.timer.get("time_zone"),
            "starting_date": self.timer.get("starting_date"),
            "starting_hour": self.timer.get("starting_hour"),
            "starting_minute": self.timer.get("starting_minute"),
            "visibility": self.visibility
        }
```